src/cbmc_starter_kit/template-for-repository/proofs/lib/summarize.py

```diff
@@ -0,0 +1,113 @@
+    return list_of_max_length_per_column
+
+
+def _get_header(list_of_max_length_per_column, header_names):
+    header = ""
+    for idx, word in enumerate(header_names):
+        max_length_of_word_in_col = list_of_max_length_per_column[idx]
+        space_formatted_word = (max_length_of_word_in_col - len(word)) * " "
+        header += "| " + word + space_formatted_word
+    header += "|\n"
+    return header
+
+
+def _get_line_separator_after_header(list_of_max_length_per_column):
+    line_sep = ""
+    for max_length_of_word_in_col in list_of_max_length_per_column:
+        line_sep += "|" + "-" * (max_length_of_word_in_col+1)
+    line_sep += "|\n"
+    return line_sep
+
+
+def _get_entries(list_of_max_length_per_column, row_data):
+    entries = []
+    for row in row_data:
+        entry = ""
+        for idx, word in enumerate(row):
+            max_length_of_word_in_col = list_of_max_length_per_column[idx]
+            space_formatted_word = (max_length_of_word_in_col - len(word)) * " "
+            entry += "| " + word + space_formatted_word
+        entry += "|\n"
+        entries.append(entry)
+    return entries
+
+
+def _get_rendered_table(data):
+    table = []
+    list_of_max_length_per_column = _get_list_of_max_length_per_column(data)
+    header = _get_header(list_of_max_length_per_column, data[0])
+    table.append(header)
+    line_sep = _get_line_separator_after_header(list_of_max_length_per_column)
+    table.append(line_sep)
+    row_data = data[1:]
+    entries = _get_entries(list_of_max_length_per_column, row_data)
+    table.extend(entries)
+    table.append("\n")
+    return "".join(table)
+
+
+def _get_status_and_proof_summaries(run_dict):
+    """Parse a dict representing a Litani run and create lists summarizing the
+    proof results.
+
+    Parameters
+    ----------
+    run_dict
+        A dictionary representing a Litani run.
+
+    Returns
+    -------
+    A list of 2 lists.
+    The first sub-list maps a status to the number of proofs with that status.
+    The second sub-list maps each proof to its status.
+    """
+    count_statuses = {}
+    proofs = [["Proof", "Status"]]
+    for proof_pipeline in run_dict["pipelines"]:
+        status_pretty_name = proof_pipeline["status"].title().replace("_", " ")
+        try:
+            count_statuses[status_pretty_name] += 1
+        except KeyError:
+            count_statuses[status_pretty_name] = 1
+        proof = proof_pipeline["name"]
+        proofs.append([proof, status_pretty_name])
+    statuses = [["Status", "Count"]]
+    for status, count in count_statuses.items():
+        statuses.append([status, str(count)])
+    return [statuses, proofs]
+
+
+def print_proof_results(out_file):
+    """
+    Print 2 strings that summarize the proof results.
+    When printing, each string will render as a GitHub flavored Markdown table.
+    """
+    with open(out_file) as fp:
+        run_dict = json.load(fp)
```

Comment:
I think this should be inside the `try`, specifically to catch `JSONDecodeError`